src/app.rs

```diff
@@ -261,28 +279,26 @@ where
     }
 
     pub fn select_prev(&self) -> anyhow::Result<()> {
-        let selected = self
-            .list_state
-            .read()
-            .selected()
-            .with_context(|| "cannot select previous item when nothing is selected")?;
+        let selected = self.list_state.read().selected().unwrap_or_else(|| 0_usize);
 
-        if selected > 0 {
+        if selected == 0 {
+            self.clear_selection()?;
+        } else if selected > 0 {
             self.select_nth(selected - 1)?;
         }
 
         Ok(())
     }
 
     pub fn select_next(&self) -> anyhow::Result<()> {
-        let selected = self
-            .list_state
-            .read()
-            .selected()
-            .with_context(|| "cannot select next item when nothing is selected")?;
+        let selected = self.list_state.read().selected();
 
-        if selected < self.items.len() - 1 {
-            self.select_nth(selected + 1)?;
+        if let Some(selected) = selected {
+            if selected < self.items.len() - 1 {
+                self.select_nth(selected + 1)?;
+            }
+        } else {
+            self.select_nth(0)?;
```

Comment:
This can apply for all, more like a UX improvement, instead of throwing error we can jump to the first item when the user selects the next item, wdyt?

Reply:
Yeah, sure, an endless list.Should definitely be optional though.